### PR TITLE
Add DEV, TEST and PROD Environment logic

### PR DESCRIPTION
**Pull request by kharyam**
Update the jenkins library function to contain workflow logic described in the tsc-docs worklfow diagram. Flows were added for DEV, TEST and PROD environments matching the TSC workflow diagram.

### CODE REVIEW

File: vars/pipelineJava8.groovy [hunk 1]
```diff
@@ -208,64 +208,219 @@ def call(
...
-          } // container
-        } // steps
-      } // stage
+      stage('DEV') {
+          when {
+              anyOf {
+                  branch 'feature/*';
+              }
+          }
+        stages {
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: nit: indentation difference

File: vars/pipelineJava8.groovy [hunk 2]
```diff
@@ -208,64 +208,219 @@ def call(
...
+
+      stage('TEST') {
+          when {
+              anyOf {
+                  branch 'master';
+                  branch 'main';
+                  branch 'hotfix/*';
+              }
+          }
+        stages {
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: nit: indentation difference

File: vars/pipelineJava8.groovy [hunk 3]
```diff
@@ -208,64 +208,219 @@ def call(
...
+      }// TEST Stage
+
+      stage('PROD') {
+          when {
+              anyOf {
+                  branch 'master';
+                  branch 'main';
+              }
+          }
+        stages {
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: nit: indentation difference

File: vars/pipelineJava8.groovy [hunk 4]
```diff
@@ -208,64 +208,219 @@ def call(
...
+
+      stage('Finish') {
+          when {
+              anyOf {
+                  branch 'master';
+                  branch 'main';
+                  branch 'hotfix/*';
+              }
+          }
+        stages {
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: nit: indentation difference

**Review by HunterGerlach — APPROVED** [review]

**Review by itewk — APPROVED** [review]

**Pull request merged**